homeassistant/components/epson/__init__.py

```diff
@@ -83,22 +86,42 @@ async def async_migrate_entry(hass, config_entry):
     """Migrate old entry."""
     _LOGGER.debug("Migrating unique_id in %s", config_entry.version)
 
-    old_unique_id = config_entry.unique_id
-    projector = hass.data[DOMAIN][config_entry.entry_id]
-    new_unique_id = await projector.get_serial_number()
-
-    @callback
-    def update_unique_id(entity_entry):
-        """Update unique ID of entity entry."""
-        return {
-            "new_unique_id": entity_entry.unique_id.replace(
-                old_unique_id, new_unique_id
+    async def get_unique_id():
+        try:
+            projector = await validate_projector(hass, config_entry.data[CONF_HOST])
+        except (CannotConnect, PoweredOff):
+            return None
+        else:
+            return await projector.get_serial_number()
+
+    if config_entry.version == 1:
+        data = {CONF_HOST: config_entry.data[CONF_HOST]}
+        hass.config_entries.async_update_entry(config_entry, data=data)
+        config_entry.version = 2
```

Comment:
I suggest we don't change config entry version since that makes it impossible to roll back. It shouldn't be needed since we don't change the config entry data structure.

Reply:
I remove `port` from config_entry.`

Reply:
An extra item doesn't matter or hurt anything.